lib/remap-lang/src/error.rs

```diff
@@ -79,31 +87,31 @@ impl fmt::Display for Rule {
             call,
             char,
             comparison,
-            EOI,
+            EOI: "",
             equality,
             expression,
             float,
             group,
-            ident,
-            if_statement,
+            ident: "",
+            if_statement: "if-statement",
             integer,
             multiplication,
-            not,
+            not: "query",
             null,
-            operator_addition,
-            operator_boolean_expr,
-            operator_comparison,
-            operator_equality,
-            operator_multiplication,
-            operator_not,
+            operator_addition: "",
+            operator_boolean_expr: "",
+            operator_comparison: "",
+            operator_equality: "",
+            operator_multiplication: "operator",
+            operator_not: "function call, value, variable, path, group, !",
             path,
-            path_coalesce,
+            path_coalesce: "coalesced path",
             path_field,
             path_index,
             path_index_inner,
             path_segment,
             path_segments,
-            primary,
+            primary: "value, variable, path, group",
```

Comment:
Unfortunately, Pest doesn't (currently) allow defining rules that are available in the parser, but transparent in error messages, which is why this rename exists.

It's less than ideal, but short of building an entire error-handling infrastructure around Pest (which we might want to do at some point, but likely not right now), or having to create a less-readable parser to make the error messages more readable, this is the best I could come up with for now.

Reply:
Yep, might be worth opening an issue there?